src/time/Waveform.hpp

```diff
@@ -48,6 +52,9 @@ class Waveform {
   /// Data values of time windows.
   Eigen::MatrixXd _timeWindows;
 
+  /// number of valid samples in _timeWindows
+  int _numberOfValidSamples;
```

Comment:
I guess we could find a better name here. The variable is only needed for extrapolation, right? Or do you see a use case for real waveform stuff as well? If not, we could use sth like `maxPossibleExtrapolationOrder`?

Reply:
`MaxPossibleExtrapolationOrder` is not ideal, because this will also be relevant for interpolation later. What we are doing is counting the number of samples that have been "locked" by the coupling scheme. This means: As soon as a time window is complete the coupling scheme will trigger the waveform to store the latest sample. This will then copy the sample to a place in `_timeWindows` where it will not change anymore. Therefore, one more sample is available for interpolation/extrapolation, when we enter the next window.

Reply:
ok, I see. But how is this then different from "numberOfTimeWindows"?

Reply:
What do you mean? Can you give a link? In `Waveform.{cpp,hpp}` I cannot find this variable.